File: Utilities/Models/Movie.cs


using System.Text.Json.Serialization;

namespace Utilities.Models;

public class Movie : BaseModel
{
    [Required]
    public int? Length { get; set; }
    [Required]
    public string? ImgURL { get; set; }
    public ICollection<Actor>? Actors { get; set; } = new List<Actor>();
    public ICollection<Theme>? Themes { get; set; } = new List<Theme>();
    [JsonIgnore]
    public ICollection<ReleaseDate>? ReleaseDate { get; set; } = new List<ReleaseDate>();
    [JsonIgnore]
    public ICollection<Hall>? Halls { get; set; } = new List<Hall>();
}

File: Client/src/app/Models/movie.ts
import { IActor } from './actor';
import { BaseModel, IBaseModel } from './baseModel';
import { IHall } from './hall';
import { ITheme } from './theme';

export interface IMovie extends IBaseModel {
  length?: number;
  release?: Date;
  actors?: IActor[];
  halls?: IHall[];
  themes?: ITheme[];
}

export class Movie extends BaseModel implements IMovie {
  length?: number;
  release?: Date;
  actors?: IActor[];
  halls?: IHall[];
  themes?: ITheme[];

  constructor() {
    super();
    this.actors = [];
    this.halls = [];
    this.themes = [];
  }
}
